lib/PersonalityProvider.jsm

```diff
@@ -71,7 +155,21 @@ this.PersonalityProvider = class PersonalityProvider {
 
   async getFromRemoteSettings(name) {
     const result = await RemoteSettings(name).get();
-    return result;
+    return Promise.all(result.map(async record => {
+      const {attachment: {filename, size}} = record;
+      // Figure out hash + md5sum check.
+      // const {attachment: {filename, hash, size}} = record;
+      // const filepath = OS.Path.join(OS.Constants.Path.localProfileDir, PERSONALITY_PROVIDER_DIR_NAME, filename);
+      const filepath = OS.Path.join(OS.Constants.Path.localProfileDir, filename);
+      // Figure out hash + md5sum check.
+      if (!await OS.File.exists(filepath) || await OS.File.stat(filepath).size !== size) {
+        await this.downloadAttachment(record);
+      }
+      return {
+        attachment: await this.getAttachment(record),
```

Comment:
There might be better a name than `attachment` here... features? classifiers? vectors? training data?